src/pyfme/aircrafts/ball.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Jan  8 18:53:17 2016
+
+@author: Juan
+
+"""
+
+import numpy as np
+from pyfme.utils.coordinates import wind2body
+from pyfme.environment.isa import atm
+
+# Constants
+mu = 1.983 * 10 ** -5  # kg/m/s
+# Data tables
+Re_list = [38000, 100000, 160000, 200000, 250000, 300000, 330000, 350000,
+           375000, 400000, 500000, 800000, 200000, 4000000]
+Cd_list = [0.49, 0.50, 0.51, 0.51, 0.49, 0.46, 0.39, 0.20, 0.09, 0.07,
+           0.07, 0.10, 0.15, 0.18]
+Sn_list = [0.00, 0.04, 0.10, 0.20, 0.40]
+Cl_list = [0.00, 0.10, 0.16, 0.23, 0.33]
+
+
+def Geometric_Data(r=0.111):
+
+    """ Provides the value of some geometric data.
+
+    Parameters
+    ----
+    r    radius(m)
+
+    References
+    ----------
+    Returns
+    ------
+    r    radius(m)
+    S_circle    Surface (m^2)
+    S_sphere    Surface (m^2)
+    Vol    Volume (m^3)
+
+    Raises
+    ------
+    See Also
+    --------
+    Notes
+    -----
+    References
+    ----------
+    .. [1]
+    """
+
+    S_circle = np.pi * r ** 2
+    S_sphere = 4 * np.pi * r ** 2
+    Vol = 4 / 3 * np.pi * r ** 3
+
+    return r, S_circle, S_sphere, Vol
+
+
+def Mass_and_Inertial_Data(r, mass=0.440):
+
+    """ Provides the value of some mass and inertial data.
+
+    Parameters
+    -----
+    r    radius (m)
+    mass   (kg)
+
+    Returns
+    ------
+    Ixxb Moment of Inertia x-axis (Kg * m2)
+    Iyyb Moment of Inertia y-axis (Kg * m2)
+    Izzb Moment of Inertia z-axis (Kg * m2)
+
+    Raises
+    ------
+    See Also
+    --------
+    Notes
+    -----
+    Ball assumed as a sphere shell
+    References
+    ----------
+    .. [1]
+    """
+
+    Ixxb = 2 * mass * (r ** 2) / 3
+    Iyyb = Ixxb
+    Izzb = Ixxb
+
+    I_matrix = np.diag([Ixxb, Iyyb, Izzb])
+
+    return I_matrix
+
+
+def Check_Reynolds_number(Re):
+    """ Reynolds number must be between 38e3 and 4e6
+    """
+    if not (Re_list[0] <= Re <= Re_list[-1]):
+        raise ValueError('Reynolds number is not inside correct range')
```

Comment:
El comportamiento por defecto de `np.interp` es usar los valores extremos cuando te sales del rango de interpolación (argumentos `left` y `right`), con lo cual si se quita el error se está usando el valor más cercano automáticamente:

http://docs.scipy.org/doc/numpy-1.10.1/reference/generated/numpy.interp.html

Por otro lado, para extrapolar hay que trabajar un poco más porque ya haría falta SciPy. Como ejercicio es interesante, pero entiendo que este código es un poco «de prueba» así que no sé si tiene mucha importancia. Como veáis.